packages/mui-material/src/Autocomplete/Autocomplete.d.ts

```diff
@@ -84,16 +85,86 @@ export interface AutocompleteRenderInputParams {
 
 export interface AutocompletePropsSizeOverrides {}
 
+export interface AutocompleteSlots {
+  /**
+   * The component used to render the listbox.
+   * @default 'ul'
+   */
+  listbox: React.JSXElementConstructor<React.HTMLAttributes<HTMLElement>>;
+  /**
+   * The component used to render the body of the popup.
+   * @default Paper
+   */
+  paper: React.JSXElementConstructor<React.HTMLAttributes<HTMLElement>>;
+  /**
+   * The component used to position the popup.
+   * @default Popper
+   */
+  popper: React.JSXElementConstructor<PopperProps>;
+}
+
+export type AutocompleteSlotsAndSlotProps<
+  Value,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined,
+  ChipComponent extends React.ElementType = ChipTypeMap['defaultComponent'],
+> = CreateSlotsAndSlotProps<
+  AutocompleteSlots,
+  {
+    chip: SlotProps<
+      React.JSXElementConstructor<ChipProps<ChipComponent>>,
+      {},
+      AutocompleteOwnerState<Value, Multiple, DisableClearable, FreeSolo, ChipComponent>
+    >;
+    clearIndicator: SlotProps<
+      React.JSXElementConstructor<Partial<IconButtonProps>>,
+      {},
+      AutocompleteOwnerState<Value, Multiple, DisableClearable, FreeSolo, ChipComponent>
+    >;
+    /**
+     * Props applied to the Listbox element.
+     */
+    listbox: SlotProps<
+      React.JSXElementConstructor<
+        ReturnType<ReturnType<typeof useAutocomplete>['getListboxProps']> & {
+          sx?: SxProps<Theme>;
+          ref?: React.Ref<Element>;
+        }
+      >,
+      {},
+      AutocompleteOwnerState<Value, Multiple, DisableClearable, FreeSolo, ChipComponent>
+    >;
+    paper: SlotProps<
```

Comment:
It looks like there is an error in the generateProptypes file, because it fails when I change to the existing types from the `ChipProps`. It only works when I use just the `ChipProps` type instead of the generic version with `ChipProps<ChipComponent>`. When I use the second one, `pnpm proptypes` fails and say the error occured in [Chip.d.ts](https://github.com/mui/material-ui/blob/next/packages/mui-material/src/Chip/Chip.d.ts#L23) because token is unsupported. I think the generateProptypes is not supporting `children?: null`.
Any ideas how to solve this?

Reply:
If we use `Partial<ChipProps>`, it will output the same as the other slotsProps, right?

Let's go for that. We're optimizing for minimal breaking changes, but I don't think this would affect many users, so I wouldn't stress about it until we get a reported issue.